dvc/fs/oss.py

```diff
@@ -8,124 +7,65 @@
-    $ export OSS_ACCESS_KEY_SECRET="my-key-secret"
-    $ export OSS_ENDPOINT="endpoint"
-    """
 
+# pylint:disable=abstract-method
+class OSSFileSystem(ObjectFSWrapper):
     scheme = Schemes.OSS
     PATH_CLS = CloudURLInfo
-    REQUIRES = {"oss2": "oss2"}
+    REQUIRES = {"ossfs": "ossfs"}
     PARAM_CHECKSUM = "etag"
-    COPY_POLL_SECONDS = 5
-    LIST_OBJECT_PAGE_SIZE = 100
-
-    def __init__(self, **config):
-        super().__init__(**config)
+    DETAIL_FIELDS = frozenset(("etag", "size"))
 
-        self.endpoint = config.get("oss_endpoint") or os.getenv("OSS_ENDPOINT")
-
-        self.key_id = (
-            config.get("oss_key_id")
-            or os.getenv("OSS_ACCESS_KEY_ID")
-            or "defaultId"
-        )
-
-        self.key_secret = (
-            config.get("oss_key_secret")
-            or os.getenv("OSS_ACCESS_KEY_SECRET")
-            or "defaultSecret"
-        )
+    def _prepare_credentials(self, **config):
+        login_info = {}
+        login_info["key"] = config.get("oss_key_id")
+        login_info["secret"] = config.get("oss_key_secret")
+        login_info["endpoint"] = config.get("oss_endpoint")
+        return login_info
 
     @wrap_prop(threading.Lock())
     @cached_property
-    def oss_service(self):
-        import oss2
-
-        logger.debug(f"key id: {self.key_id}")
-        logger.debug(f"key secret: {self.key_secret}")
-
-        return oss2.Auth(self.key_id, self.key_secret)
-
-    def _get_bucket(self, bucket):
-        import oss2
+    def fs(self):
+        from ossfs import OSSFileSystem as _OSSFileSystem
 
-        return oss2.Bucket(self.oss_service, self.endpoint, bucket)
+        return _OSSFileSystem(**self.fs_args)
 
-    def _generate_download_url(self, path_info, expires=3600):
-        return self._get_bucket(path_info.bucket).sign_url(
-            "GET", path_info.path, expires
-        )
-
-    def exists(self, path_info) -> bool:
-        paths = self._list_paths(path_info)
-        return any(path_info.path == path for path in paths)
-
-    def _list_paths(self, path_info):
-        import oss2
-
-        for blob in oss2.ObjectIterator(
-            self._get_bucket(path_info.bucket), prefix=path_info.path
-        ):
-            yield blob.key
-
-    def walk_files(self, path_info, **kwargs):
-        if not kwargs.pop("prefix", False):
-            path_info = path_info / ""
-        for fname in self._list_paths(path_info):
-            if fname.endswith("/"):
-                continue
-
-            yield path_info.replace(path=fname)
-
-    def remove(self, path_info):
-        if path_info.scheme != self.scheme:
-            raise NotImplementedError
-
-        logger.debug(f"Removing oss://{path_info}")
-        self._get_bucket(path_info.bucket).delete_object(path_info.path)
-
-    def _upload_fobj(self, fobj, to_info, **kwargs):
-        self._get_bucket(to_info.bucket).put_object(to_info.path, fobj)
+    def _upload_fobj(self, fobj, to_info, size=None):
```

Comment:
There are some differences between the oss-emulator and the real servier. about 40% percent of the methods are unsupported in the emulator. If we do not use the emulator, these lines can be removed.

Reply:
@karajan1001 Is this particular method require all of those methods to be implemented? Or are there any quick fixes for the emulator to get rid of the workaround logic?

Reply:
@efiop , Sorry not quite understand. Most of 40% percent of the methods which oss-emulator didn't support are not used in dvc. But this one in the base class would use them, and this would cause tests failure in `Linux`. And the `Macos` and `Windows` tests which uses real server would not be affected. If we replace the emulator with real server in `Linux` tests. We can pass it.

If you mean to repair the lacking maintenance `oss-emulator`. It requires weeks or even months of work.

Reply:
@karajan1001  I thought that there is one or two broken/missing methods in emulator that we need to fix to make it work in these simple func tests, but if it will take so much time then yeah, let's just use a real thing for now. I would keep emulator fixtures too though (similar to s3 and real_s3 fixtures that we have), so that if someone will be willing to fix the emulator - he could in the future. Also, as a part of moving to plugins, we'll likely get rid of many oss tests from dvc itself in the future, so we'll later figure out what to do with those fixtures.